### DISCUSSION TITLE
Redirect Uri when editing client in TUI

### DISCUSSION THREAD

**Discussion by wlio-tom** [JanssenProject/jans#9511] [Q&A] [answered]
I set up a client in the TUI with 3 redirect Uris, ex. subdomain.domain.com/callback. Later I made a change as subdomain.domain.com/login/callback. I now get the error redirect uri does not match a previously registered client (my front end logic gives the updated uri).

I checked the logs and found the redirect uri comparison was still being made to the old values, yet the TUI and database show the updated uri. I have re deployed but not deleted the database at this point.

Could this be a bug or have I missed a step after updating redirect Uris?

**Comment by ossdhaval** [comment]
Hi @wlio-tom 

We are checking. In the meanwhile, [here](https://docs.jans.io/v1.1.5/admin/auth-server/client-management/client-configuration/#redirect-uri) is the useful documentation around redirect URIs and how Janssen Server compares and manages it.

Adding @yuriyz for his comments.

**Reply by ossdhaval** [reply]
Hi @wlio-tom We checked. But authentication is getting through if the request contains the new URI. 

Can you ensure that the URI in the auth request is the updated one? Can you share the screenshot of the TUI client details page and HAR of the authentication request that you are sending.

**Reply by wlio-tom** [reply]
Hi @ossdhaval Thanks for taking a look.

<img width="762" alt="Screenshot 2024-09-17 at 6 56 48 PM" src="https://github.com/user-attachments/assets/34434d69-960f-436a-bd6f-4370b5f5902d">

<img width="861" alt="image" src="https://github.com/user-attachments/assets/1e0d20a9-687b-4472-bb64-7107e9a1b41f">

<img width="1710" alt="image" src="https://github.com/user-attachments/assets/37751630-ca9b-4f74-9b5d-2cae8bdbf100">

What is a little unusual is that I did not actually add these redirect Uris for this client, they were from a previously registered client.

**Reply by yuriyz** [reply, marked as answer]
According to log there is no `https://advisor.wealthinc.app/login/callback` registered with client. 
1. Check sector identifier. If you have it set, redirect uris are read out of sector identifier.
2. Set logging level to trace, it will give you more information what is going on.

**Reply by wlio-tom** [reply]
It looks like my sector identifier was the issue.
When I change sector-identifier.json do I need to take any additional steps?
Logs show comparisons are still the old values, but switching Subject Type to Public resolves the problem.

**Reply by yuriyz** [reply]
Sector identifier response is cached for 1d. If you fix your sector identifier redirect uris, it will automatically get updated in 1d. Or you can force update by AS restart.

There is `sectorIdentifierCacheLifetimeInMinutes` AS configuration property which allows to change cache lifetime.

**Reply by wlio-tom** [reply]
I waited out the cache and now fully resolved. Thanks for your help!

**Reply by yuriyz** [reply]
Great, thanks for info.
